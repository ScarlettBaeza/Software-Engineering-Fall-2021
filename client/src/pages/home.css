
.text {
    position: absolute;
    color: white;
    top: 45%;
    left: 50%;
    transform: translate(-50%, -50%);
}

.sugma {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}
