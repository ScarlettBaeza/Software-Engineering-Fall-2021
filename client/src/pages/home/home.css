@import url(https://fonts.googleapis.com/css?family=Roboto:400,300,500);

.resMake {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
}

#loginBox {
    position: relative;
    margin: 5% auto;
    width: 600px;
    height: 400px;
    background: #FFF;
    border-radius: 2px;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
  }

.left {
    position: absolute;
    top: 10;
    padding: 40px;
    width: 300px;
    height: 400px;
  }
  
  h1 {
    margin: 0 0 20px 0;
    font-weight: 300;
    font-size: 28px;
  }

  .right {
    position: absolute;
    top: 10;
    right: 0;
    padding: 40px;
    width: 300px;
    height: 400px;
    border-radius: 0 2px 2px 0;
  }

  .or {
    position: absolute;
    top: 180px;
    left: 280px;
    width: 40px;
    height: 40px;
    background: #DDD;
    border-radius: 50%;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.4);
    line-height: 40px;
    text-align: center;
  }

  input[type="text"],
  input[type="password"] {
    display: block;
    box-sizing: border-box;
    margin-bottom: 20px;
    padding: 4px;
    width: 220px;
    height: 32px;
    border: none;
    border-bottom: 1px solid #5e5e5e;
    font-family: 'Roboto', sans-serif;
    font-weight: 400;
    font-size: 15px;
    transition: 0.2s ease;
  }

  .left .signup {
    display: block;
    margin-bottom: 40px;
    font-size: 28px;
    color: rgb(0, 0, 0);
    text-align: center;
  }

  .right .contGuest {
    display: block;
    margin-bottom: 40px;
    font-size: 28px;
    color: rgb(0, 0, 0);
    text-align: center;
  }