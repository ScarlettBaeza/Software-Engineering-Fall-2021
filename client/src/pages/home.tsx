import React from 'react';
import Button from 'react-bootstrap/Button';
import background from '../assets/images/background.jpg'
import axios from 'axios';
import './home.css';
import Reservation from '../models/reservation';

export const Home = () => {
    const handleTest = () => {
        const testReservation = new Reservation(123,"dan",12,undefined)

        axios.post("http://localhost:8080/reservation", testReservation)
        .then((res) => console.log(res.data));

       
    }

    const handleGet = () => {
        axios.get('http://localhost:8080/reservation')
        .then((result)=>console.log(result.data))
    }

    return (
        <>
            <img
            src = {background}
            alt = 'Background'/>
            <div className = "text">EAT NOW</div>
            <Button className = "sugma" onClick = {handleTest}>Make a Reservation</Button>
            <Button className = "testy" onClick = {handleGet}> Get All Reservations</Button>
        </>
    );
}