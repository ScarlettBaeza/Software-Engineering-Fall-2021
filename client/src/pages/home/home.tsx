import React from 'react';
import Button from 'react-bootstrap/Button';
import axios from 'axios';
import './home.css';
import Reservation from '../../models/reservation';
import Table from '../../models/table'

export const Home = () => {
    return (
        <>
        <div id="loginBox">
            <div className="left">
                <h1 className="signup">Sign up Today</h1>
                <input className="text" name="username" placeholder="Username" />
                <input className="text" name="email" placeholder="E-mail" />
                <input type="password" name="password" placeholder="Password" />
                <input type="password" name="password2" placeholder="Retype password" />
                <Button type="submit" name="signup_submit" value="Sign me up" href="registeredReserve"> Sign me up </Button>
            </div> 

            <div className="or">OR</div>
            
            <div className="right">
                <h1 className="contGuest">Continue as Guest</h1>
                <Button className = "resMake" href="resForm">Make a Reservation</Button>
            </div>
        </div>
        </>
    );
}