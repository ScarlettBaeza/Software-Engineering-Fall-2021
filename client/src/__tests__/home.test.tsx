import React from 'react';
import renderer from 'react-test-renderer';
import {Home} from '../pages/home/home';
import '@testing-library/jest-dom';

it(' home renders correctly', () => {
    const tree = renderer
      .create(<Home/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
