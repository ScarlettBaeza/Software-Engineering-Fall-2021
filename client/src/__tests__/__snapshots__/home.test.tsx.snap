// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` home renders correctly 1`] = `
<div
  id="loginBox"
>
  <div
    className="left"
  >
    <h1
      className="signup"
    >
      Sign up Today
    </h1>
    <input
      className="text"
      name="username"
      placeholder="Username"
    />
    <input
      className="text"
      name="email"
      placeholder="E-mail"
    />
    <input
      name="password"
      placeholder="Password"
      type="password"
    />
    <input
      name="password2"
      placeholder="Retype password"
      type="password"
    />
    <button
      className="btn btn-primary"
      disabled={false}
      name="signup_submit"
      type="submit"
      value="Sign me up"
    >
       Sign me up 
    </button>
  </div>
  <div
    className="or"
  >
    OR
  </div>
  <div
    className="right"
  >
    <h1
      className="contGuest"
    >
      Continue as Guest
    </h1>
    <a
      className="resMake btn btn-primary"
      href="resForm"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Make a Reservation
    </a>
  </div>
</div>
`;
